### PR TITLE
feat: [PL-40790]: add filters for Notifications in Default Settings

### PR DESCRIPTION
**Pull request by namanharness**
#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 1]
```diff
@@ -105,45 +106,179 @@ DefaultSettingsFactory.registerCategory('NOTIFICATIONS', {
...
 
+    return (
+      <DefaultSettingsTagInput
+        {...props}
+        tagInputProps={{
+          tagsProps: {
+            separator: defaultSettingsTagInputSeparator,
+            onChange: values => onSettingSelectionChange(values.filter(Boolean).join(',')),
+            values: settingValue?.value?.split(defaultSettingsTagInputSeparator) ?? [],
+            placeholder: 'Example: abc123xyz',
```
Comment (vikrant-harness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:271]: move this to getString

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 2]
```diff
@@ -105,45 +106,179 @@ DefaultSettingsFactory.registerCategory('NOTIFICATIONS', {
...
+
+    return (
+      <DefaultSettingsTagInput
+        {...props}
+        tagInputProps={{
+          tagsProps: {
+            separator: defaultSettingsTagInputSeparator,
+            onChange: values => onSettingSelectionChange(values.filter(Boolean).join(',')),
+            values: settingValue?.value?.split(defaultSettingsTagInputSeparator) ?? [],
+            placeholder: 'Example: webhook.office.com',
```
Comment (vikrant-harness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:240]: move this to getString

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 3]
```diff
@@ -105,45 +106,179 @@ DefaultSettingsFactory.registerCategory('NOTIFICATIONS', {
...
+DefaultSettingsFactory.registerSettingHandler(SettingType.SLACK_NOTIFICATION_ENDPOINTS_ALLOWLIST, {
+  label: 'platform.defaultSettings.addSlackNotificationFilterValues',
+  settingRenderer: props => {
+    const { onSettingSelectionChange, settingValue } = props
+
+    return (
+      <DefaultSettingsTagInput
+        {...props}
+        tagInputProps={{
+          tagsProps: {
```
Comment (vikrant-harness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:174]: can we extract this out to a common function that receives the placeholder and returns this JSX , it is repeated 4 times

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 4]
```diff
@@ -105,45 +106,179 @@ DefaultSettingsFactory.registerCategory('NOTIFICATIONS', {
...
+            separator: defaultSettingsTagInputSeparator,
+            onChange: values => onSettingSelectionChange(values.filter(Boolean).join(',')),
+            values: settingValue?.value?.split(defaultSettingsTagInputSeparator) ?? [],
+            placeholder: 'Example: webhook.slack.com',
```
Comment (vikrant-harness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:178]: move this to getString

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 5]
```diff
@@ -105,45 +106,179 @@ DefaultSettingsFactory.registerCategory('NOTIFICATIONS', {
...
+
+    return (
+      <DefaultSettingsTagInput
+        {...props}
+        tagInputProps={{
+          tagsProps: {
+            separator: defaultSettingsTagInputSeparator,
+            onChange: values => onSettingSelectionChange(values.filter(Boolean).join(',')),
+            values: settingValue?.value?.split(defaultSettingsTagInputSeparator) ?? [],
+            placeholder: 'Example: webhook.company.com',
```
Comment (vikrant-harness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:209]: move this to getString

**Review by vikrant-harness — DISMISSED** [review]

File: src/modules/27-platform/default-settings/components/ReusableHandlers.tsx [hunk 6]
```diff
@@ -221,8 +221,24 @@ export const DefaultSettingsToggle: React.FC<SettingRendererProps> = ({
...
-export const DefaultSettingsTagInput = ({ identifier, tagInputProps }: DefaultSettingsTagInputProps): JSX.Element => {
-  return <FormInput.KVTagInput className={css.defaultSettingRenderer} name={identifier} {...tagInputProps} />
+export const DefaultSettingsTagInput = ({
+  identifier,
+  onSettingSelectionChange,
+  settingValue,
+  getString,
+  placeholder
+}: DefaultSettingsTagInputProps): JSX.Element => {
+  const defaultSettingsTagInputSeparator = ','
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/components/ReusableHandlers.tsx:234]: Use this from src/modules/27-platform/default-settings/utils/utils.ts

File: src/modules/27-platform/default-settings/interfaces/SettingType.types.ts [hunk 7]
```diff
@@ -13,7 +13,10 @@ export enum SettingGroups {
   test_group_2 = 'test_group_2',
   PERSPECTIVES_PREFERENCES = 'perspectives_preferences',
   TICKETING_PREFERENCES = 'ticketing_preferences',
-  ENABLED_NOTIFICATION_CHANNELS_GROUP = 'enabled_notification_channels'
+  SLACK_NOTIFICATION_SETTINGS_GROUP = 'slack_notification_settings',
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/interfaces/SettingType.types.ts:16]:
```suggestion
  SLACK_NOTIFICATION_SETTINGS = 'slack_notification_settings',
```

File: src/modules/27-platform/default-settings/interfaces/SettingType.types.ts [hunk 8]
```diff
@@ -13,7 +13,10 @@ export enum SettingGroups {
...
+  MSTEAM_NOTIFICATION_SETTINGS_GROUP = 'msTeam_notification_settings',
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/interfaces/SettingType.types.ts:17]:
```suggestion
  MSTEAM_NOTIFICATION_SETTINGS = 'msTeam_notification_settings',
```

File: src/modules/27-platform/default-settings/interfaces/SettingType.types.ts [hunk 9]
```diff
@@ -13,7 +13,10 @@ export enum SettingGroups {
...
+  WEBHOOK_NOTIFICATION_SETTINGS_GROUP = 'webhook_notification_settings',
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/interfaces/SettingType.types.ts:18]:
```suggestion
  WEBHOOK_NOTIFICATION_SETTINGS = 'webhook_notification_settings',
```

File: src/modules/27-platform/default-settings/interfaces/SettingType.types.ts [hunk 10]
```diff
@@ -13,7 +13,10 @@ export enum SettingGroups {
...
+  PAGERDUTY_NOTIFICATION_SETTINGS_GROUP = 'pagerduty_notification_settings'
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/interfaces/SettingType.types.ts:19]:
```suggestion
  PAGERDUTY_NOTIFICATION_SETTINGS = 'pagerduty_notification_settings'
```

File: src/modules/27-platform/default-settings/components/ReusableHandlers.tsx [hunk 11]
```diff
@@ -221,8 +221,24 @@ export const DefaultSettingsToggle: React.FC<SettingRendererProps> = ({
 
 export interface DefaultSettingsTagInputProps extends SettingRendererProps {
   tagInputProps?: Omit<KVTagInputProps, 'name'>
+  placeholder: keyof StringsMap
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/components/ReusableHandlers.tsx:224]:
```suggestion
  placeholder: StringKeys
```

File: src/modules/27-platform/default-settings/components/ReusableHandlers.tsx [hunk 12]
```diff
@@ -221,8 +221,24 @@ export const DefaultSettingsToggle: React.FC<SettingRendererProps> = ({
...
+  identifier,
+  onSettingSelectionChange,
+  settingValue,
+  getString,
+  placeholder
+}: DefaultSettingsTagInputProps): JSX.Element => {
+  const defaultSettingsTagInputSeparator = ','
+
+  const tagInputProps = {
+    defaultSettingsTagInputSeparator,
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/default-settings/components/ReusableHandlers.tsx:237]: In Earlier code, the key name for this prop was simply `separator`. 
It was passed as a prop to `<FormInput.KVTagInput />` inside tagsProps.
The type for tagsProp is `ITagInputProps` (https://blueprintjs.com/docs/versions/3/#core/components/tag-input) which takes `separator` as a prop name.

Please re-check this once.

Comment (namanharness) [reply src/modules/27-platform/default-settings/components/ReusableHandlers.tsx:237]: i have checked this for existing use cases as well in Default Settings, the APIs currently take `,` as separator so it is
working for all cases, in future if it is changed we can update the component, generally it is `,` only

**Review by vikrant-harness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**